### PR TITLE
fix overlay transform query parameter parsing bug

### PR DESCRIPTION
**Pull request by cmac1000**
refactors query parameter parsing logic to make it the
responsibility of the concrete transform classes, because
one type of delimiter doesn't work for all of them.

ref: #140

### CODE REVIEW

File: its/tests/test_overlay.py [hunk 1]
```diff
@@ -16,6 +16,18 @@ def test_simple_overlay(self):
         )
         assert response.status_code == 200
 
+    def test_simple_overlay_with_underscores(self):
+        response = self.client.get(
+            "tests/images/test.png?overlay=tests/images/five-something_with_underscores.png"
+        )
+        assert response.status_code == 200
+
+    def test_empty_overlay(self):
```
Comment (wamberg) [inline its/tests/test_overlay.py]: What about when `parameters < 1`? I pushed ab08e4d760 with this test. There's an issue here, but it's also in production:

https://image.pbs.org/contentchannels/2714/14jWbEcLCgV84jJS7fDA.jpg?overlay=

Do you think fixing this bug is in the scope of this PR? If so, can you take a shot?

Comment (cmac1000) [reply its/tests/test_overlay.py]: ha, good one. yeah, should be an easy fix...

Comment (cmac1000) [reply its/tests/test_overlay.py]: I think it's a 400 in that situation though: https://github.com/pbs/its/pull/142/commits/c494e3d7705ab90a203e93583a2e8cded799b800

File: its/transformations/overlay.py [hunk 2]
```diff
@@ -23,13 +24,21 @@ class OverlayTransform(BaseTransform):
...
+
         if len(parameters) > 1:
-            LOGGER.error(
-                "overlay transform does not accept parameters, received %s", parameters
-            )
+            raise ValueError("overlay transform does not accept parameters")
 
         overlay = parameters[0]
+        if not overlay:
+            raise ITSClientError("no overlay image supplied")
```
Comment (wamberg) [inline its/transformations/overlay.py]: I don't think we're getting here in the empty parameters scenario. I've updated `test_empty_overlay` to make it fail.

Comment (wamberg) [reply its/transformations/overlay.py]: On second thought, I'm not sure I updated the test correctly. I wanted to test that we were actually raising an `ITSClientError` and not an `IndexError` on `overlay = parameters[0]` where `parameters = []`.

Do you know a better way to test that we're running `raise ITSClientError("no overlay image supplied")`?

**Pull request merged**